nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,11 +423,215 @@ async fn run() -> Result<()> {
+**/
+fn get_winners(
+    tasks: &Vec<AssignerTask>,
+    result: Option<String>
+) -> (Option<ResultPayload>, Vec<Sender>) {
+    tracing::debug!("Result is: {:?}", result);
+    let mut winners: Vec<Sender> = vec![];
+    let mut result_payload: Option<ResultPayload> = None;
+    tasks.iter().for_each(|task| {
+        let task_result = get_task_identifier(task);
+        if task_result == result && !task_result.is_none() {
+            winners.push(task.worker.clone());
+            result_payload = Some(task.result.clone());
+        }
+    });
+
+    (result_payload, winners)
+}
+
+// Return values from a task identifier, in case of
+// failure happened, return None
+fn get_task_identifier(
+    task: &AssignerTask
+) -> Option<String> {
+    match get_result_identifier(task.result.output.clone()) {
+        Ok(Some(val)) => {
+            Some(val)
+        }
+        Ok(None) => {
+            None
+        }
+        Err(err) => {
+            tracing::debug!("Failed to parse identifier, error: {}", err);
+            None
+        }
+    }
+}
+
+async fn reward(
+    assign_event_id: String,
+    workers: &Vec<Sender>,
+    payload: ResultPayload,
+    classifier_result_collection: &Collection<ClassifierResult>,
+    collection: &Collection<FinishedJobs>,
+    job_type: u16,
+) -> bool {
+    tracing::debug!(
+        "All the results for {} are consistent, sending",
+        assign_event_id
+    );
+    let raw_data_id = payload.header.raw_data_id.clone();
+    let result_identifier = get_result_identifier(payload.output.clone());
+    match result_identifier   {
+        Err(_) =>{},
+        Ok(None)=>{},
+        Ok(Some(tag_id)) =>{
+            // Save classifier result to db
+            let classifier_result = ClassifierResult {
+                _id: raw_data_id,
+                kv_key: payload.kv_key.clone(),
+                tag_id
+            };
+            match classifier_result_collection.insert_one(classifier_result, None).await {
+                Ok(_) => {
+                    tracing::info!("Write classifier result to db: {}", payload.kv_key.clone());
+                }
+                Err(err) => {
+                    tracing::error!("Cannot write classifier result to db: {}", err);
+                }
+            }
+        }
+    }
+
+    let db_entry = FinishedJobs {
+        _id: raw_data_id,
+        workers: workers.clone(), // if we remove first, no need to clone but this is safer
+        timestamp: chrono::Utc::now().timestamp() as u64,
+        result: payload,
+        job_type,
+        assign_event_id,
+    };
+    tracing::debug!(
+        "Rewarding all workers for data: {}",
+        raw_data_id.to_string()
+    );
+    match collection.insert_one(db_entry, None).await {
+        Ok(_) => {
+            FINISHED_JOBS.inc();
+            true
+        }
+        Err(err) => {
+            // We log and that's it, we keep the entry in book for later inspection
```

Comment:
This comment is from the old implementation, I don't think it's relevant any more